--- a/Makefile	2023-01-07 21:53:00.000000000 +0800
+++ b/Makefile	2023-01-29 14:44:37.312461505 +0800
@@ -837,6 +837,9 @@
 else ifdef CONFIG_CC_OPTIMIZE_FOR_SIZE
 KBUILD_CFLAGS += -Os
 KBUILD_RUSTFLAGS += -Copt-level=s
+else ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE_O3
+KBUILD_CFLAGS += -O3
+KBUILD_RUSTFLAGS += -Copt-level=3
 endif
 
 # Always set `debug-assertions` and `overflow-checks` because their default
