--- a/arch/x86/Makefile	2022-02-18 19:34:07.037907718 +0800
+++ b/arch/x86/Makefile	2022-02-18 19:35:13.168928355 +0800
@@ -126,7 +126,7 @@
                 $(call cc-option,-march=core2,$(call cc-option,-mtune=generic))
 	cflags-$(CONFIG_MATOM) += $(call cc-option,-march=atom) \
 		$(call cc-option,-mtune=atom,$(call cc-option,-mtune=generic))
-        cflags-$(CONFIG_GENERIC_CPU) += $(call cc-option,-mtune=generic)
+        cflags-$(CONFIG_GENERIC_CPU) += $(call cc-option,-mtune=sandybridge)
         KBUILD_CFLAGS += $(cflags-y)
 
         KBUILD_CFLAGS += -mno-red-zone
@@ -204,6 +204,9 @@
 LDFLAGS_vmlinux :=
 endif
 
+LDFLAGS_vmlinux += $(call ld-option,--relax)
+LDFLAGS_vmlinux += -O2
+
 #
 # The 64-bit kernel must be aligned to 2MB.  Pass -z max-page-size=0x200000 to
 # the linker to force 2MB page size regardless of the default page size used
