--- arch/x86/Makefile	2023-10-03 20:39:16.803332147 +0800
+++ arch/x86/Makefile	2023-10-03 20:36:52.731900898 +0800
@@ -62,7 +62,7 @@
 
 # More compiler optimizations
 ifdef CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE
-KBUILD_CFLAGS += -O3 -fexcess-precision=fast -freciprocal-math \
+KBUILD_CFLAGS += -O3 -freciprocal-math \
 		 $(call cc-option,-fcx-limited-range)
 KBUILD_RUSTFLAGS += -Copt-level=3
 endif
@@ -200,7 +200,7 @@
   KBUILD_CFLAGS += -mharden-sls=all
 endif
 
-KBUILD_LDFLAGS += -m elf_$(UTS_MACHINE)
+KBUILD_LDFLAGS += -m elf_$(UTS_MACHINE) -O3
 
 ifdef CONFIG_LTO_CLANG
 ifeq ($(shell test $(CONFIG_LLD_VERSION) -lt 130000; echo $$?),0)
@@ -220,7 +220,7 @@
 # by the linker.
 #
 ifdef CONFIG_X86_64
-LDFLAGS_vmlinux += -z max-page-size=0x200000
+LDFLAGS_vmlinux += -z max-page-size=0x200000 -O3 --relax $(call ld-option, --icf=safe)
 endif
 
 
