--- a/net/ipv4/tcp_westwood.c	2022-10-30 22:12:07.000000000 +0800
+++ b/net/ipv4/tcp_westwood.c	2022-10-30 22:14:55.195343887 +0800
@@ -44,8 +44,8 @@
 };
 
 /* TCP Westwood functions and constants */
-#define TCP_WESTWOOD_RTT_MIN   (HZ/20)	/* 50ms */
-#define TCP_WESTWOOD_INIT_RTT  (20*HZ)	/* maybe too conservative?! */
+#define TCP_WESTWOOD_RTT_MIN   (HZ/50)	/* 20ms */
+#define TCP_WESTWOOD_INIT_RTT  (1*HZ)	/* maybe not too conservative?! */
 
 /*
  * @tcp_westwood_create
