--- a/include/net/tcp.h	2026-03-27 06:42:42.268811465 +0000
+++ b/include/net/tcp.h	2026-03-27 06:41:23.686767730 +0000
@@ -87,16 +87,16 @@
 #define TCP_BASE_MSS		1024
 
 /* probing interval, default to 10 minutes as per RFC4821 */
-#define TCP_PROBE_INTERVAL	600
+#define TCP_PROBE_INTERVAL	720
 
 /* Specify interval when tcp mtu probing will stop */
-#define TCP_PROBE_THRESHOLD	8
+#define TCP_PROBE_THRESHOLD	6
 
 /* After receiving this amount of duplicate ACKs fast retransmit starts. */
 #define TCP_FASTRETRANS_THRESH 3
 
 /* Maximal number of ACKs sent quickly to accelerate slow-start. */
-#define TCP_MAX_QUICKACKS	16U
+#define TCP_MAX_QUICKACKS	18U
 
 /* Maximal number of window scale according to RFC1323 */
 #define TCP_MAX_WSCALE		14U
@@ -139,7 +139,7 @@
 				 * initial RTO.
 				 */
 
-#define TCP_TIMEWAIT_LEN (60*HZ) /* how long to wait to destroy TIME-WAIT
+#define TCP_TIMEWAIT_LEN (56*HZ) /* how long to wait to destroy TIME-WAIT
 				  * state, about 60 seconds	*/
 #define TCP_FIN_TIMEOUT	TCP_TIMEWAIT_LEN
                                  /* BSD style FIN_WAIT2 deadlock breaker.
@@ -149,25 +149,25 @@
 				  */
 #define TCP_FIN_TIMEOUT_MAX (120 * HZ) /* max TCP_LINGER2 value (two minutes) */
 
-#define TCP_DELACK_MAX	((unsigned)(HZ/5))	/* maximal time to delay before sending an ACK */
+#define TCP_DELACK_MAX	((unsigned)(HZ/8))	/* maximal time to delay before sending an ACK */
 static_assert((1 << ATO_BITS) > TCP_DELACK_MAX);
 
 #if HZ >= 100
-#define TCP_DELACK_MIN	((unsigned)(HZ/25))	/* minimal time to delay before sending an ACK */
-#define TCP_ATO_MIN	((unsigned)(HZ/25))
+#define TCP_DELACK_MIN	((unsigned)(HZ/38))	/* minimal time to delay before sending an ACK */
+#define TCP_ATO_MIN	((unsigned)(HZ/38))
 #else
 #define TCP_DELACK_MIN	4U
 #define TCP_ATO_MIN	4U
 #endif
-#define TCP_RTO_MAX_SEC 120
+#define TCP_RTO_MAX_SEC 66
 #define TCP_RTO_MAX	((unsigned)(TCP_RTO_MAX_SEC * HZ))
-#define TCP_RTO_MIN	((unsigned)(HZ / 5))
+#define TCP_RTO_MIN	((unsigned)(HZ / 10))
 #define TCP_TIMEOUT_MIN	(2U) /* Min timeout for TCP timers in jiffies */
 
 #define TCP_TIMEOUT_MIN_US (2*USEC_PER_MSEC) /* Min TCP timeout in microsecs */
 
-#define TCP_TIMEOUT_INIT ((unsigned)(1*HZ))	/* RFC6298 2.1 initial RTO value	*/
-#define TCP_TIMEOUT_FALLBACK ((unsigned)(3*HZ))	/* RFC 1122 initial RTO value, now
+#define TCP_TIMEOUT_INIT ((unsigned)(HZ/5*2))	/* RFC6298 2.1 initial RTO value	*/
+#define TCP_TIMEOUT_FALLBACK ((unsigned)(HZ/5*6))	/* RFC 1122 initial RTO value, now
 						 * used as a fallback RTO for the
 						 * initial data transmission if no
 						 * valid RTT sample has been acquired,
@@ -268,7 +268,7 @@
 #define TCP_THIN_LINEAR_RETRIES 6       /* After 6 linear retries, do exp. backoff */
 
 /* TCP initial congestion window as per rfc6928 */
-#define TCP_INIT_CWND		10
+#define TCP_INIT_CWND		12
 
 /* Bit Flags for sysctl_tcp_fastopen */
 #define	TFO_CLIENT_ENABLE	1
@@ -838,7 +838,7 @@
 
 static inline u32 __tcp_set_rto(const struct tcp_sock *tp)
 {
-	return usecs_to_jiffies((tp->srtt_us >> 3) + tp->rttvar_us);
+	return max(usecs_to_jiffies(((tp->srtt_us >> 3) + tp->rttvar_us) * 9 / 10), READ_ONCE(tp->inet_conn.icsk_rto_min));
 }
 
 u32 tcp_delack_max(const struct sock *sk);
