--- a/arch/x86/boot/compressed/Makefile	2024-12-09 16:55:55.000000000 +0000
+++ b/arch/x86/boot/compressed/Makefile	2024-12-31 11:53:44.240055870 +0000
@@ -24,7 +24,7 @@
 # case of cross compiling, as it has the '--target=' flag, which is needed to
 # avoid errors with '-march=i386', and future flags may depend on the target to
 # be valid.
-KBUILD_CFLAGS := -m$(BITS) -O2 $(CLANG_FLAGS)
+KBUILD_CFLAGS := -m$(BITS) -Os $(CLANG_FLAGS)
 KBUILD_CFLAGS += -fno-strict-aliasing -fPIE
 KBUILD_CFLAGS += -Wundef
 KBUILD_CFLAGS += -DDISABLE_BRANCH_PROFILING

--- a/arch/x86/entry/vdso/Makefile	2024-12-31 11:57:02.053142438 +0000
+++ b/arch/x86/entry/vdso/Makefile	2024-12-31 11:57:08.969950940 +0000
@@ -51,7 +51,7 @@
 # Don't omit frame pointers for ease of userspace debugging, but do
 # optimize sibling calls.
 #
-CFL := $(PROFILING) -mcmodel=small -fPIC -O2 -fasynchronous-unwind-tables -m64 \
+CFL := $(PROFILING) -mcmodel=small -fPIC -O3 -fasynchronous-unwind-tables -m64 \
        $(filter -g%,$(KBUILD_CFLAGS)) -fno-stack-protector \
        -fno-omit-frame-pointer -foptimize-sibling-calls \
        -DDISABLE_BRANCH_PROFILING -DBUILD_VDSO
