--- a/Makefile	2024-06-21 21:16:57.366643142 +0000
+++ b/Makefile	2024-06-21 21:17:28.830703245 +0000
@@ -1109,7 +1109,7 @@
 CC_FLAGS_LTO	+= -fvisibility=hidden
 
 # Limit inlining across translation units to reduce binary size
-KBUILD_LDFLAGS += -mllvm -import-instr-limit=5
+KBUILD_LDFLAGS += -mllvm -import-instr-limit=5 --lto-O3
 
 # Check for frame size exceeding threshold during prolog/epilog insertion
 # when using lld < 13.0.0.
