--- a/kernel/cpu.c	2024-05-03 14:48:56.241774489 +0800
+++ b/kernel/cpu.c	2024-05-03 14:58:29.833080238 +0800
@@ -37,6 +37,7 @@
 #include <linux/cpuset.h>
 #include <linux/random.h>
 #include <linux/cc_platform.h>
+#include <linux/interrupt.h>
 
 #include <trace/events/power.h>
 #define CREATE_TRACE_POINTS
@@ -1300,6 +1301,9 @@
 	enum cpuhp_state target = max((int)st->target, CPUHP_AP_OFFLINE);
 	int err, cpu = smp_processor_id();
 
+	if(local_softirq_pending())
+		return -16;
+
 	/* Ensure this CPU doesn't handle any more interrupts. */
 	err = __cpu_disable();
 	if (err < 0)
