--- a/lib/Kconfig.debug	2024-12-30 09:56:33.674179613 +0000
+++ b/lib/Kconfig.debug	2024-12-30 09:35:32.183197909 +0000
@@ -525,6 +525,13 @@
 
 	  It is mainly for debug and performance tuning use.
 
+config DEBUG_FORCE_FUNCTION_ALIGN_32B
+        bool "Force all function address 32B aligned"
+        depends on EXPERT && (X86_64 || ARM64 || PPC32 || PPC64 || ARC || RISCV || S390) && (!DEBUG_FORCE_FUNCTION_ALIGN_64B)
+        select FUNCTION_ALIGNMENT_32B
+        help
+          32 bit alignment
+
 #
 # Select this config option from the architecture Kconfig, if it
 # is preferred to always offer frame pointers as a config

--- a/arch/x86/Makefile	2024-12-31 06:35:00.287699967 +0000
+++ b/arch/x86/Makefile	2024-12-31 06:35:10.032404710 +0000
@@ -157,10 +157,10 @@
         KBUILD_CFLAGS += -m64
 
         # Align jump targets to 1 byte, not the default 16 bytes:
-        KBUILD_CFLAGS += $(call cc-option,-falign-jumps=1)
+        KBUILD_CFLAGS += $(call cc-option,-falign-jumps=4)
 
         # Pack loops tightly as well:
-        KBUILD_CFLAGS += $(call cc-option,-falign-loops=1)
+        KBUILD_CFLAGS += $(call cc-option,-falign-loops=4)
 
         # Don't autogenerate traditional x87 instructions
         KBUILD_CFLAGS += -mno-80387

--- a/lib/Kconfig.debug   2024-12-30 09:35:32.183197909 +0000
+++ b/lib/Kconfig.debug  2024-12-31 11:29:37.320048523 +0000
@@ -532,6 +532,12 @@
         help
           32 bit alignment

+config DEBUG_FORCE_FUNCTION_ALIGN_32B_SOFT
+       bool "allow smaller fallback alignment by compiler choice"
+       depends on DEBUG_FORCE_FUNCTION_ALIGN_32B && (!CFI_CLANG)
+       help
+         saves space slightly for some functions
+
 #
 # Select this config option from the architecture Kconfig, if it
 # is preferred to always offer frame pointers as a config

--- a/Makefile
+++ b/Makefile
@@ -879,7 +879,7 @@
 
 # Clear used registers at func exit (to reduce data lifetime and ROP gadgets).
 ifdef CONFIG_ZERO_CALL_USED_REGS
-KBUILD_CFLAGS	+= -fzero-call-used-regs=used-gpr
+KBUILD_CFLAGS	+= -fzero-call-used-regs=used-gpr-arg
 endif
 
 ifdef CONFIG_FUNCTION_TRACER
@@ -982,12 +982,21 @@
 export CC_FLAGS_NO_FPU
 
 ifneq ($(CONFIG_FUNCTION_ALIGNMENT),0)
-# Set the minimal function alignment. Use the newer GCC option
-# -fmin-function-alignment if it is available, or fall back to -falign-funtions.
-# See also CONFIG_CC_HAS_SANE_FUNCTION_ALIGNMENT.
-ifdef CONFIG_CC_HAS_MIN_FUNCTION_ALIGNMENT
-KBUILD_CFLAGS += -fmin-function-alignment=$(CONFIG_FUNCTION_ALIGNMENT)
-else
-KBUILD_CFLAGS += -falign-functions=$(CONFIG_FUNCTION_ALIGNMENT)
-endif
+ifeq ($(CONFIG_DEBUG_FORCE_FUNCTION_ALIGN_32B_SOFT),y)
+CONFIG_FUNCTION_ALIGNMENT = 24
+endif
+REAL_FUNCTION_ALIGNMENT = $(CONFIG_FUNCTION_ALIGNMENT)
+ifeq ($(REAL_FUNCTION_ALIGNMENT), 24)
+REAL_FUNCTION_ALIGNMENT = 32:19:16:12
+endif
+ifeq ($(call cc-option,-falign-functions=$(REAL_FUNCTION_ALIGNMENT)),)
+REAL_FUNCTION_ALIGNMENT = $(CONFIG_FUNCTION_ALIGNMENT)
+endif
+KBUILD_CFLAGS += -falign-functions=$(REAL_FUNCTION_ALIGNMENT)
+ifeq ($(shell expr $(CONFIG_FUNCTION_ALIGNMENT) \> 8), 1)
+KBUILD_CFLAGS += $(call cc-option, -fmin-function-alignment=8)
+endif
+ifeq ($(shell expr $(CONFIG_FUNCTION_ALIGNMENT) \> 23), 1)
+KBUILD_CFLAGS += $(call cc-option, -flimit-function-alignment)
+endif
 endif
