--- a/drivers/firmware/efi/libstub/Makefile	2024-05-03 07:57:05.000000000 +0800
+++ b/drivers/firmware/efi/libstub/Makefile	2024-05-11 17:29:04.483951136 +0800
@@ -106,7 +106,7 @@
 # .note.gnu.property for code-less object files (like lib/ctype.c),
 # so work around this by explicitly removing the unwanted section.
 # https://bugs.llvm.org/show_bug.cgi?id=46480
-STUBCOPY_FLAGS-y		+= --remove-section=.note.gnu.property
+STUBCOPY_FLAGS-y		+= --remove-section=.note.gnu.property --remove-section=.gnu.build.attributes
 
 STUBCOPY_RELOC-$(CONFIG_X86_32)	:= R_386_32
 STUBCOPY_RELOC-$(CONFIG_X86_64)	:= R_X86_64_64
@@ -162,7 +162,7 @@
 #
 quiet_cmd_stubcopy = STUBCPY $@
       cmd_stubcopy =							\
-	$(STRIP) --strip-debug -o $@ $<;				\
+	$(STRIP) -R .gnu.build.attributes --strip-debug -o $@ $<;				\
 	if $(OBJDUMP) -r $@ | grep $(STUBCOPY_RELOC-y); then		\
 		echo "$@: absolute symbol references not allowed in the EFI stub" >&2; \
 		/bin/false;						\
