diff --git a/net/ipv4/tcp_input.c b/net/ipv4/tcp_input.c
index a9ac5fa06d4a..8fa9150d6cf7 100644
--- a/net/ipv4/tcp_input.c
+++ b/net/ipv4/tcp_input.c
@@ -3953,6 +3953,34 @@ static u32 tcp_newly_delivered(struct sock *sk, u32 prior_delivered, int flag)
 	return delivered;
 }
 
+static void tcp_check_local_cc(struct sock *sk)
+{
+    struct tcp_sock *tp = tcp_sk(sk);
+    const struct inet_connection_sock *icsk = inet_csk(sk);
+    const struct dst_entry *dst;
+    u32 min_rtt;
+    min_rtt = tcp_min_rtt(tp);
+
+    if (unlikely(min_rtt == 0))
+        return;
+
+    if (unlikely(icsk->icsk_ca_setsockopt))
+        return;
+
+    dst = __sk_dst_get(sk);
+    if (dst && dst_metric(dst, RTAX_CC_ALGO))
+        return;
+
+    if (likely(READ_ONCE(tp->bytes_acked) > 0x200000))
+        return;
+
+    if (unlikely(min_rtt <= 750)) {
+        if (unlikely(strncmp(icsk->icsk_ca_ops->name, "highspeed", 9) != 0)) {
+            tcp_set_congestion_control(sk, "highspeed", false, true);
+        }
+    }
+}
+
 /* This routine deals with incoming acks, but not outgoing ones. */
 static int tcp_ack(struct sock *sk, const struct sk_buff *skb, int flag)
 {
@@ -4075,6 +4103,8 @@ static int tcp_ack(struct sock *sk, const struct sk_buff *skb, int flag)
 	flag |= tcp_clean_rtx_queue(sk, skb, prior_fack, prior_snd_una,
 				    &sack_state, flag & FLAG_ECE);
 
+	tcp_check_local_cc(sk);
+
 	tcp_rack_update_reo_wnd(sk, &rs);
 
 	tcp_in_ack_event(sk, flag);
